tuf/data/interface.go

```diff
@@ -0,0 +1,28 @@
+package data
+
+// TufRole is an interface for dealing with TUF roles internally
+type TufRole interface {
+	// Returns true for BaseRole
+	IsBaseRole() bool
```

Comment:
good point about casting being more of a golang way of doing things

I think the only place we may want this shared interface other than signing (where we have good amount of context, as @endophage commented) is when dealing with targets/delegations since a good amount of that code is intertwined, but casting should be able to work fine.  I'll update this